File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
  let check = 0; 
    for(let x of s){
      if(x==="(") check ++
      else if(x===")") check --//여기서 삼항연산자 (x==="(")? check ++:check-- 쓰면 시간초과 
      if(check < 0)return false
    }
    if(check === 0)return true;
  	else return false
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque

def solution(s):
    open = []
    data = deque(s)
    while data:
        tmp = data.popleft()
        if(tmp == "("):
            open.append(tmp)
        else:
            if(len(open)>0):
                open.pop()
            else: return False
    else: 
        if(len(open)) > 0: return False
    return True 